python/ray/dashboard/modules/metrics/metrics_head.py

```diff
@@ -73,6 +76,10 @@ def __init__(self, dashboard_head):
         self.prometheus_host = os.environ.get(
             PROMETHEUS_HOST_ENV_VAR, DEFAULT_PROMETHEUS_HOST
         )
+        self.prometheus_headers = os.environ.get(
```

Comment:
can we do some basic validation of the format before using the env var?

Reply:
A new `parse_prom_headers` function is added for both validation and parsing. This is what it looks like when the validation fails:

```bash
▶ export RAY_PROMETHEUS_HEADERS='[["XAuth"],["XAuth2"]]'
▶ ray start --head --include-dashboard=true --metrics-export-port=8080

Local node IP: 127.0.0.1
2024-12-28 13:32:56,919	ERROR services.py:1353 -- Failed to start the dashboard , return code 1
2024-12-28 13:32:56,920	ERROR services.py:1378 -- Error should be written to 'dashboard.log' or 'dashboard.err'. We are printing the last 20 lines for you. See 'https://docs.ray.io/en/master/ray-observability/user-guides/configure-logging.html#logging-directory-structure' to find where the log file is.
Traceback (most recent call last):
  ...
  File "/Users/ruian/Code/python/ray/python/ray/dashboard/modules/metrics/metrics_head.py", line 97, in __init__
    self.prometheus_headers = parse_prom_headers(
                              ^^^^^^^^^^^^^^^^^^^
  File "/Users/ruian/Code/python/ray/python/ray/dashboard/modules/metrics/metrics_head.py", line 74, in parse_prom_headers
    raise ValueError(
ValueError: RAY_PROMETHEUS_HEADERS should be a JSON string in one of the formats:
1) An object with string keys and string values.
2) an array of string arrays with 2 string elements each.
```